src/PrivateDns/PrivateDns/Models/PrivateDnsClient.cs

```diff
@@ -401,6 +400,13 @@ private static void FillEmptyRecordsForType(RecordSet properties, RecordType rec
 
         public PSPrivateDnsRecordSet UpdatePrivateDnsRecordSet(PSPrivateDnsRecordSet recordSet, bool overwrite)
         {
+            var getRecordSet = GetPrivateDnsRecordSet(recordSet.Name, recordSet.ZoneName, recordSet.ResourceGroupName,
```

Comment:
Dont you want to account for the case where this retrieved record set may be null?

Reply:
added null check.